app/instance-initializers/head.js

```diff
@@ -1,30 +1,33 @@
 import ENV from '../config/environment';
 
 export function initialize(owner) {
-  if (ENV['ember-cli-head'] && ENV['ember-cli-head']['suppressBrowserRender']) { return true; }
+  if (typeof FastBoot === 'undefined') {
+    if (ENV['ember-cli-head'] && ENV['ember-cli-head']['suppressBrowserRender']) { return true; }
 
-  // clear fast booted head (if any)
-  let startMeta = document.querySelector('meta[name="ember-cli-head-start"]')
-  let endMeta = document.querySelector('meta[name="ember-cli-head-end"]')
-  if (startMeta && endMeta) {
-    let el = startMeta.nextSibling
-    while(el && el !== endMeta) {
-      document.head.removeChild(el);
-      el = startMeta.nextSibling;
+    // clear fast booted head (if any)
+    let startMeta = document.querySelector('meta[name="ember-cli-head-start"]')
+    let endMeta = document.querySelector('meta[name="ember-cli-head-end"]')
+    if (startMeta && endMeta) {
+      let el = startMeta.nextSibling
+      while(el && el !== endMeta) {
+        document.head.removeChild(el);
+        el = startMeta.nextSibling;
+      }
+      document.head.removeChild(startMeta);
+      document.head.removeChild(endMeta);
     }
-    document.head.removeChild(startMeta);
-    document.head.removeChild(endMeta);
-  }
 
-  const component = owner.lookup('component:head-layout');
-  component.appendTo(document.head);
+    const component = owner.lookup('component:head-layout');
+    component.appendTo(document.head);
+  } else {
```

Comment:
> I'm really confused on what the suggested way to handle fastboot/browser code is. Originally it was separate builds, then it was a single build with flags guarding, now it seems there will be separate builds again?

There are still no separate builds now. We are building the additional asset containing such fastboot overrides. For the fastboot initializers we never asked to guard against the flags. It's only for the browser initializers that you need to move from `app/initializers/browser/somefile.js` to `app/initializers/somefile.js` and put the guard. For the `app/initializers/fastboot/somefile.js` to `fastboot/initializers/somefile.js`.

We are working on updating the fastboot website to talk about the new scheme and new structure. There is also a cheatsheet that talks about the various migration cases.